berkdb/mp/mp_fopen.c

```diff
@@ -873,11 +878,13 @@ __memp_fopen(dbmfp, mfp, path, flags, mode, pagesize)
 		if (dbmfp->ftype != 0)
 			mfp->ftype = dbmfp->ftype;
 
+		found = 1;
```

Comment:
how about using a new variable loopmfp for the for loop, and mfp can be function scope--no need for found:
if (!fileid_mpf) {
   MPOOLFILE *loop_mfp;
   for_each_(....,loop_mfp, ..) {
     ...
    mfp = loop_mfp; // no need for found = 1;
    break;
    }
}